src/python/pants/core_tasks/generate_pants_ini.py

```diff
@@ -0,0 +1,60 @@
+# coding=utf-8
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import configparser
+import logging
+import os
+import sys
+
+from pants.base.exceptions import TaskError
+from pants.task.task import Task
+from pants.version import VERSION
+
+
+logger = logging.getLogger(__name__)
+
+
+class GeneratePantsIni(Task):
+  """Generate pants.ini with sensible defaults."""
+
+  PANTS_INI = "pants.ini"
+
+  class CustomConfigParser(configparser.ConfigParser):
+    """We monkey-patch the write() function to allow us to write entries in the style `key: value`, 
+       rather than `key:value` or `key : value`."""
+
+    def write(self, fp):
+      # Refer to https://github.com/jaraco/configparser/blob/294a985b759bddd3505ace54ecf56da39c56a302/src/backports/configparser/__init__.py#L916
+      # for the original source.
+      delimiter = ": "
+      if self._defaults:
+        self._write_section(fp, self.default_section, self._defaults.items(), delimiter)
+      for section in self._sections:
+        self._write_section(fp, section, self._sections[section].items(), delimiter)
+
+  def execute(self):
+    if os.stat(self.PANTS_INI).st_size != 0:
+      raise TaskError("{} is not empty! This goal is only meant for first-time "
+                      "users. Please update config values by directly modifying the file.".format(self.PANTS_INI))
+
+    pants_version = VERSION
+    python_version = ".".join(str(v) for v in sys.version_info[:2])
+
+    config = self.CustomConfigParser()
+    logger.info("Adding sensible defaults to the `{}` config file:".format(self.PANTS_INI))
```

Comment:
That said re logging - this probsably ought to be a `ConsoleTask`:

```python
def console_output(self, _):
  ...
  config = self.CustomConfigParser()
  yield "Adding sensible defaults to the `{}` config file:".format(self.PANTS_INI)
  
  yield "* Pinning `pants_version` to `{}`.".format(pants_version)
  config["GLOBAL"] = {"pants_version": pants_version}
  ...
```

Reply:
I'm not sure of the provenance of the Console Task doc string warning, but it's no worse to modify pants.ini than in a regular Task. Either way you're mutating the source of options which you've already had read for you upstream to even be able to construct your task. The idea of this being a console task is also correct in v2 where console rules are the only rules users can directly invoke from the cli.